doc/user/content/guides/node-js.md

```diff
@@ -0,0 +1,245 @@
+```
+
+The code above opens a long-lived transaction with `BEGIN` and uses [`TAIL` with `FETCH`](/sql/tail/#tailing-with-fetch) to receive a stream of updates to `my_view`. The [TAIL Output format](/sql/tail/#output) of `res.rows` is an array of view update objects.
+
+**Note on output format:** When a row of a tailed view is **updated,** two objects will show up in the `rows` array:
+
+```js
+[
+    ...
+    {
+        mz_timestamp: '1627225629000',
+        mz_diff: '1',
+        my_column_one: 'ABC',
+        my_column_two: 'new_value'
+    },
+    {
+        mz_timestamp: '1627225629000',
+        mz_diff: '-1',
+        my_column_one: 'ABC',
+        my_column_two: 'old_value'
+    },
+    ...
+]
+```
+
+An `mz_diff` value of `-1` indicates Materialize is deleting one row with the included values.  An update is just a deletion (`mz_diff: '-1'`) and an insertion (`mz_diff: '1'`) with the same `mz_timestamp`.
+
+<!--- NOT PUBLISHABLE UNTIL https://github.com/brianc/node-postgres/pull/2573 is merged
+### Using pg-query-stream
+
+```js
+const { Client } = require('pg');
+const QueryStream = require('pg-query-stream');
+
+const client = new Client('postgres://materialize@localhost:6875/materialize');
+
+client.connect((err, client) => {
+  if (err) {
+    throw err;
+  }
+  const stream = client.query(new QueryStream('TAIL avg_bid', []));
+  stream.pipe(process.stdout);
+});
+```
+--->
+
+## Query
+
+Querying Materialize is identical to querying a traditional PostgreSQL database: Node.js executes the query, and Materialize returns the state of the view, source, or table at that point in time.
+
+Query a view `my_view` with a select statement:
+
+```js
+const { Client } = require('pg');
+const client = new Client('postgres://materialize@localhost:6875/materialize');
+
+async function main() {
+  await client.connect();
+  const res = await client.query('SELECT * FROM my_view');
+  console.log(res.rows);
+};
+
+main();
+```
+
+Polling _(repeatedly querying)_ of a view is fine, and response times are much faster than traditional database queries because Materialize maintains materialized views in memory. For more, refer to the node-postgres documentation of [Queries in node-postgres](https://node-postgres.com/features/queries) and [pg.Result](https://node-postgres.com/api/result) for output reference.
+
+**Note:** If your SELECT queries do more than read out of a materialization, think about whether you should materialize part of the query, or the whole thing.
+
+## Push data to a source
+
+Data produced in a Node.js app **cannot be sent directly to a [`SOURCE` in Materialize](/sql/create-source/)** for a couple of reasons:
+
+-  Materialize relies on another system (a traditional database, S3 or Kafka) to serve as the "system of record" for data. Materialize processes a live stream of data and maintains views in memory.
+-  Sending data to an intermediary system helps decouple the operation of the Node.js app from the operation of Materialize.
```

Comment:
The second point actually isn't a reason why data can't be sent directly; it's just a beneficial result.

This introduction is where we should clarify what sources do, instead of repeating it multiple times in the table below. Something like:

> Materialize processes live streams of data and maintains views in memory, relying on other systems (traditional databases, S3, or Kafka) to serve as "systems of record" for the data. Instead of updating Materialize directly, Node.js sends the data to an *intermediary system*. Your Materialize installation connects to the intermediary system as a "source" and reads streaming updates from that.

Intermediary systems supported for Node.js are: